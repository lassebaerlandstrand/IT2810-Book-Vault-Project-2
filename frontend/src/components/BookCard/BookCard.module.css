.card {
  height: 100%;
  min-height: 20rem;
}

:where([data-mantine-color-scheme='light']) .card {
  background-color: var(--mantine-color-gray-0);
}

:where([data-mantine-color-scheme='light']) .bookAuthor {
  color: var(--mantine-color-gray-7);
}

:where([data-mantine-color-scheme='dark']) .bookAuthor {
  color: var(--mantine-color-dark-1);
}

.bookTitle {
  font-size: var(--mantine-font-size-md);
}

.bookTitle,
.bookAuthor {
  text-align: center;
  text-overflow: ellipsis;
  overflow-wrap: break-word;
  overflow: hidden;
}

.ratingSection {
  display: flex;
  justify-content: center;
}

@media screen and (min-width: $mantine-breakpoint-xxs) {
  .bookTitle,
  .bookAuthor {
    text-align: start;
  }

  .ratingSection {
    justify-content: left;
  }
}
