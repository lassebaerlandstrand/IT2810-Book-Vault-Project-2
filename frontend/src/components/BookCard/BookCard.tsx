import { Card, Image, Text } from '@mantine/core';
import { Book } from '@/generated/graphql';
import styles from './BookCard.module.css';

type BookCardProps = {
  book: Book;
};

const BookCard = ({ book }: BookCardProps) => {
  return (
    <>
      <Card p={30} radius="lg" className={styles.card} m="auto">
        <Card.Section h={200}>
          <Image
            src={book.coverImg}
            alt={`Cover image for ${book.title}`}
            fallbackSrc="https://placehold.co/200x300?text=Cover%20image%20for%20book"
            radius="lg"
            fit="contain"
            w="fit-content"
            maw="100%"
            mah="100%"
            m="auto"
          />
        </Card.Section>
        <Card.Section mt="xs" h={50} className={styles.textSection}>
          <Text fw={600} component="h4" lineClamp={2} title={book.title}>
            {book.title}
          </Text>
          <Text size="xs" c="dimmed" lineClamp={2} title={book.authors.join(', ')}>
            {book.authors[0]} {book.authors.length > 1 ? 'et al.' : ''}
          </Text>
        </Card.Section>
      </Card>
    </>
  );
};

export default BookCard;
