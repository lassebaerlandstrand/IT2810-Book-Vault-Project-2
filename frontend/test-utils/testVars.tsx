import { Book } from '@/generated/graphql';

export const dummyBook: Book = {
  __typename: 'Book',
  authors: [{ __typename: 'Author', name: 'A name' }],
  coverImg: 'https://placehold.co/200x300?text=Cover%20image%20for%20book',
  genres: [{ __typename: 'Genre', name: 'A genre' }],
  id: '1',
  publisher: { __typename: 'Publisher', name: 'A publisher' },
  title: 'A title',
  bookFormat: 'Paper',
  characters: ['A character'],
  description: 'A deep book about deep topics',
  isbn: '999999999',
  language: 'Swedish',
  pages: 1000,
  publishDate: '03/03/03',
  rating: 3.555,
  series: 'Title #1',
  numRatings: 5000,
  ratingsByStars: [1, 5, 10, 7, 1],
};
