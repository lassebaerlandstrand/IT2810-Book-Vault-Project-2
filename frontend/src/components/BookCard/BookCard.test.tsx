import { render } from '@test-utils';
import { dummyBook } from '../../../test-utils/testVars';
import BookCard from './BookCard';

describe('Bookcard component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<BookCard book={dummyBook} />);
    const attributesToRemove = [
      ...document.body.querySelectorAll('div [id^="mantine"]'),
      ...document.body.querySelectorAll('div [for^="mantine"]'),
    ]; // Because Mantine uses random ids which causes snapshots to fail
    attributesToRemove.forEach((element) => {
      element.removeAttribute('for');
      element.removeAttribute('id');
      element.removeAttribute('class');
      element.removeAttribute('aria-describedby');
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
