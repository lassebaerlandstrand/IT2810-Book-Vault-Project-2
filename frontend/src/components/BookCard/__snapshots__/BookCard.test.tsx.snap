// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Bookcard component > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: Poppins, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-font-family-headings: Poppins, Roboto, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji;--mantine-breakpoint-xs: 30em;--mantine-breakpoint-md: 64em;--mantine-breakpoint-xl: 90em;--mantine-breakpoint-xxs: 24em;--mantine-color-logo-theme-0: #fefefe;--mantine-color-logo-theme-1: #cbcfd2;--mantine-color-logo-theme-2: #97a0a6;--mantine-color-logo-theme-3: #7d8991;--mantine-color-logo-theme-4: #63717b;--mantine-color-logo-theme-5: #495a65;--mantine-color-logo-theme-6: #2f434f;--mantine-color-logo-theme-7: #132937;--mantine-color-logo-theme-8: #0b202e;--mantine-color-logo-theme-9: #061017;}:root[data-mantine-color-scheme="dark"]{--mantine-color-logo-theme-text: var(--mantine-color-logo-theme-4);--mantine-color-logo-theme-filled: var(--mantine-color-logo-theme-8);--mantine-color-logo-theme-filled-hover: var(--mantine-color-logo-theme-9);--mantine-color-logo-theme-light: rgba(47, 67, 79, 0.15);--mantine-color-logo-theme-light-hover: rgba(47, 67, 79, 0.2);--mantine-color-logo-theme-light-color: var(--mantine-color-logo-theme-3);--mantine-color-logo-theme-outline: var(--mantine-color-logo-theme-4);--mantine-color-logo-theme-outline-hover: rgba(99, 113, 123, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-logo-theme-text: var(--mantine-color-logo-theme-filled);--mantine-color-logo-theme-filled: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-filled-hover: var(--mantine-color-logo-theme-7);--mantine-color-logo-theme-light: rgba(47, 67, 79, 0.1);--mantine-color-logo-theme-light-hover: rgba(47, 67, 79, 0.12);--mantine-color-logo-theme-light-color: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-outline: var(--mantine-color-logo-theme-6);--mantine-color-logo-theme-outline-hover: rgba(47, 67, 79, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 29.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 30em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 63.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 64em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 89.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 90em) {.mantine-hidden-from-xl {display: none !important;}}@media (max-width: 23.99375em) {.mantine-visible-from-xxs {display: none !important;}}@media (min-width: 24em) {.mantine-hidden-from-xxs {display: none !important;}}
  </style>
  <div
    class="_card_0962fe m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
    style="--paper-radius: var(--mantine-radius-lg); margin: auto;"
  >
    <div
      class="m_599a2148 mantine-Card-section"
      data-first-section="true"
    >
      <img
        alt="Cover image for A title"
        class="m_9e117634 mantine-Image-root"
        src="https://placehold.co/200x300?text=Cover%20image%20for%20book"
        style="--image-radius: var(--mantine-radius-lg); --image-object-fit: contain; margin: auto; max-width: 100%; max-height: 100%;"
      />
    </div>
    <div
      class="m_599a2148 mantine-Card-section _textSection_0962fe"
      data-last-section="true"
    >
      <h4
        class="mantine-focus-auto _bookTitle_0962fe m_b6d8b162 mantine-Text-root"
        data-line-clamp="true"
        style="--text-line-clamp: 2; font-weight: 600;"
        title="A title"
      >
        A title
      </h4>
      <p
        class="mantine-focus-auto _bookAuthor_0962fe m_b6d8b162 mantine-Text-root"
        data-line-clamp="true"
        data-size="xs"
        style="--text-fz: var(--mantine-font-size-xs); --text-lh: var(--mantine-line-height-xs); --text-line-clamp: 2;"
        title="A name"
      >
        A name 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
